File: common/src/main/java/org/sandboxpowered/silica/network/handshake/clientbound/StatusRequest.java
package org.sandboxpowered.silica.network.handshake.clientbound;

import org.sandboxpowered.silica.network.*;
import org.sandboxpowered.silica.network.handshake.serverbound.StatusResponse;

public class StatusRequest implements Packet {
    @Override
    public void read(PacketByteBuf buf) {

    }

    @Override
    public void write(PacketByteBuf buf) {

    }

    @Override
    public void handle(PacketHandler packetHandler, Connection connection) {
        packetHandler.sendPacket(new StatusResponse());
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/vanilla/network/handshake/clientbound/StatusRequest.kt
package org.sandboxpowered.silica.vanilla.network.handshake.clientbound

import org.sandboxpowered.silica.vanilla.network.Connection
import org.sandboxpowered.silica.vanilla.network.Packet
import org.sandboxpowered.silica.vanilla.network.PacketByteBuf
import org.sandboxpowered.silica.vanilla.network.PacketHandler
import org.sandboxpowered.silica.vanilla.network.handshake.serverbound.StatusResponse

class StatusRequest() : Packet {
    constructor(buf: PacketByteBuf) : this()

    override fun write(buf: PacketByteBuf) {}

    override fun handle(packetHandler: PacketHandler, connection: Connection) {
        packetHandler.sendPacket(StatusResponse(connection.motd))
    }
}